syntax = "proto2";

package module;

option java_multiple_files = true;
option java_package = "proto.module";
option java_outer_classname = "ModuleProto";

message EmptyMessage{

}

message MessageRequest{
  required int64 idSender = 1;
  required string message = 2;
}

message Module{
  required string type = 1;
  required int64 id = 2;
}

message WelcomeRequest{
  required module.Module module = 1;
}

message GoodByeRequest{
  required module.Module module = 1;
}

service ModuleService{
  rpc receiveMessage(module.MessageRequest) returns (module.EmptyMessage);
  rpc close(module.EmptyMessage) returns (module.EmptyMessage);
  rpc welcome(module.WelcomeRequest) returns (module.EmptyMessage);
  rpc goodBye(module.GoodByeRequest) returns (module.EmptyMessage);
}

